src/vs/base/parts/ipc/common/ipc.net.ts

```diff
@@ -151,11 +151,11 @@ export const enum ProtocolConstants {
 	/**
 	 * If there is no reconnection within this time-frame, consider the connection permanently closed...
 	 */
-	ReconnectionGraceTime = 3 * 60 * 60 * 1000, // 3hrs
+	ReconnectionGraceTime = 5000, // 5 seconds
```

Comment:
We don't want to change this as it would break things in remote which depend on these long timeouts, instead let's pass special reconnection times defined somewhere within the platform/terminal into PersistentTerminal